core/cap-0035.md

```diff
@@ -61,37 +61,104 @@ exchange of assets, throughout the globe, enabling users to make payments betwee
-from the designated account based on the amount provided in the `ClawbackOp`
+This proposal introduces new `ClawbackOp` and `ClawbackClaimableBalancesOp`
+operations.
+
+The `AUTH_CLAWBACK_ENABLED_FLAG` flag on the issuing account must be set when a
+trustline is created to authorize a `ClawbackOp` operation submitted by the
+issuer account.
+
+A claimable balance inherits its clawback enabled status from the account
+creating it and a `ClawbackClaimableBalanceOp` operation is only valid for
+claimable balances created from accounts that have clawback enabled.
+
+The `ClawbackOp` and `ClawbackClaimableBalanceOp` operations result in the
+removal of the specified assets issued from the network.
+
+The `ClawbackOp` and `ClawbackClaimableBalanceOp` operations only apply to
+assets issued by the source account. Assets that can be clawed back are easily
+distinguished from traditional blockchain assets (bearer instruments) so that
+asset owners are aware of rights.
+
+The `ClawbackOp` may result in revocation of some or all of the specified
+assets from the designated account based on the amount provided in the
 operation.
 
 ## Specification
 
 ### XDR changes
 
-This patch of XDR changes is based on the XDR files in commit `f60c329467f5f637b4137ca220fde4e609434f53` of [stellar-core].
+This patch of XDR changes is based on the XDR files in tag/commit `v15.1.0` (`90b2780584c6390207bf09291212d606896ce9f8`) of [stellar-core].
 
 ```diff
+diff --git a/src/xdr/Stellar-ledger-entries.x b/src/xdr/Stellar-ledger-entries.x
+index 8d746391..3f887e79 100644
+--- a/src/xdr/Stellar-ledger-entries.x
++++ b/src/xdr/Stellar-ledger-entries.x
+@@ -99,7 +99,9 @@ enum AccountFlags
+     // otherwise, authorization cannot be revoked
+     AUTH_REVOCABLE_FLAG = 0x2,
+     // Once set, causes all AUTH_* flags to be read-only
+-    AUTH_IMMUTABLE_FLAG = 0x4
++    AUTH_IMMUTABLE_FLAG = 0x4,
++    // Trustlines are created with clawback enabled set to "true"
++    AUTH_CLAWBACK_ENABLED_FLAG = 0x8
+ };
+ 
+ // mask for all valid flags
+@@ -187,12 +189,16 @@ enum TrustLineFlags
+     AUTHORIZED_FLAG = 1,
+     // issuer has authorized account to maintain and reduce liabilities for its
+     // credit
+-    AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG = 2
++    AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG = 2,
++    // issuer has specified that it may clawback its credit, and that claimable
++    // balances created with its credit must be claimable by the issuer
++    CLAWBACK_ENABLED = 4
+ };
+ 
+ // mask for all trustline flags
+ const MASK_TRUSTLINE_FLAGS = 1;
+ const MASK_TRUSTLINE_FLAGS_V13 = 3;
++const MASK_TRUSTLINE_FLAGS_V16 = 7;
+ 
+ struct TrustLineEntry
+ {
+@@ -291,7 +297,8 @@ enum ClaimPredicateType
+     CLAIM_PREDICATE_OR = 2,
+     CLAIM_PREDICATE_NOT = 3,
+     CLAIM_PREDICATE_BEFORE_ABSOLUTE_TIME = 4,
+-    CLAIM_PREDICATE_BEFORE_RELATIVE_TIME = 5
++    CLAIM_PREDICATE_BEFORE_RELATIVE_TIME = 5,
++    CLAIM_PREDICATE_UNCONDITIONAL_CLAWBACK = 6

```

Comment:
@jonjove @sisuresh @MonsieurNicolas Instead of introducing the issuer as a claimant with a predicate, would you be okay with the claimable balance being given a flags where the one flag would be `CLAWBACK_ENABLED`?